src/data/mv_polynomial/basic.lean

```diff
@@ -557,6 +560,9 @@ add_monoid_algebra.support_mul_single p _ (by simp) _
   (X s * p).support = p.support.map (add_left_embedding (single s 1)) :=
 add_monoid_algebra.support_single_mul p _ (by simp) _
 
+@[simp] lemma support_smul_eq [no_zero_divisors R] {a : R} (h: a≠0) {p: mv_polynomial σ R}:
+(a • p).support = p.support := finsupp.support_smul_eq h
```

Comment:
General style comments - spaces around colons and indent wrapped lines
```suggestion
@[simp] lemma support_smul_eq [no_zero_divisors R] {a : R} (h : a≠0) (p : mv_polynomial σ R) :
  (a • p).support = p.support := finsupp.support_smul_eq h
```

Reply:
Thanks! And sorry for my carelessness in style.
I had thought an explicit `p` would be better, but I saw that `finsupp.support_smul_eq` has an implicit argument in the place of `p`, so I followed it.

Reply:
Should this lemma be changed to a more general version as follows?
```lean
@[simp] lemma support_smul_eq {S₁: Type*} [semiring S₁] [module S₁ R] [no_zero_smul_divisors S₁ R]
  {a : S₁} (h : a≠0) (p : mv_polynomial σ R) : (a • p).support = p.support :=
finsupp.support_smul_eq h
```

Reply:
That sounds good to me; though please add spaces around `≠`!